### DISCUSSION TITLE
Determining when to use FileOpenPicker or IFileDialog

### DISCUSSION THREAD

**Discussion by lb90** [microsoft/WindowsAppSDK#3779] [Q&A] [answered]
As reported in https://learn.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker#in-a-desktop-app-that-requires-elevation (see also https://github.com/microsoft/WindowsAppSDK/issues/2504), apps should use the WinRT file pickers when run non-elevated and the Common Item Dialogs (IFileDialog) when run elevated. Which method should be used to determine if running elevated or not?

* Should we use [PrivilegeCheck](https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-privilegecheck)?
* Should we use [GetTokenInformation](https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-gettokeninformation), similarly to the [AppContainer](https://learn.microsoft.com/en-us/windows/win32/secauthz/appcontainer-for-legacy-applications-#c-canonical) check?
* Any other method?

Thank you very much!
Luca

**Comment by DHancock** [comment, marked as answer]
If you require an unmanged version, the runtime installer (and others) use this:

https://github.com/microsoft/WindowsAppSDK/blob/main/dev/Common/Security.IntegrityLevel.h

**Reply by Gaoyifei1011** [reply]
Is there a c# version of this code?

**Reply by DHancock** [reply]
````c#
    private static bool GetIsElevated()
    {
        try
        {
            using (WindowsIdentity identity = WindowsIdentity.GetCurrent())
            {
                WindowsPrincipal principal = new WindowsPrincipal(identity);
                return principal.IsInRole(WindowsBuiltInRole.Administrator);
            }
        }
        catch (Exception ex)
        {
            Debug.Fail(ex.ToString());
        }

        return false;
    }
````
(with the caveat that I haven't tested it in a domain environment)

**Reply by Gaoyifei1011** [reply]
Thanks for your answer, this code solves my problem
